### DISCUSSION TITLE
CC2652P7 Support

### DISCUSSION THREAD

**Discussion by tube0013** [Koenkk/Z-Stack-firmware#470] [Ideas]
Hi,

there are now 2652P7 modules available from RF-Star, and the CC1352P7 works on them with no issue, however the PA is not utilized and it's capped at TX = 5. TI's recommendation is to start with a p7 port over the application: 

> The CC13x2x7 or CC26x2x7 is pin compatible with, and functionally identical to, their respective CC13x2x1 or CC26x2x1 devices but with an additional 352 kB of flash memory and extra 64 kB of RAM. However, it is not currently possible to use the Hardware View tool to change the device types of projects using CC13x2x1 or CC26x2x1 to CC13x2x7 or CC26x2x7. Developers are strongly encouraged to import a project which contains the correct settings for their target device and port the stack application accordingly.

How feasible would it be to pull in the PA settings from the znp_CC1352P_2_LAUNCHXL project?

**Comment by Koenkk** [comment]
I'm a bit confused by TIs recommendation. From my understanding they mean: compile the CC1352P7 SDK (which I do). This fw can also be used on the CC2652P7 (there is no separate project for this):

<img width="392" alt="Screenshot 2023-08-09 at 19 27 15" src="https://github.com/Koenkk/Z-Stack-firmware/assets/2892853/c2456183-7c0d-493b-a8ef-c30dca15639a">

The CC1352P7 firmware I provide can do up to TX = 9

**Comment by tube0013** [comment]
Yes, I just did a complie with 9. My question/requestt is the RF-Star CC2652P7 based module is essentially the same as the P2, they just swapped the chips, so it has the same PA set up which was compatible with the LaunchPad, so could that be ported to the CC1352P7 FW.  Probably will wait and see what happens in future sdk versions too I guess.

**Reply by Koenkk** [reply]
Ah I see, so a custom firmware is required for this (which is a pitty). I can provide these FWs but only once adapters are commercially/easily available, I don't want to maintain too many firmwares.

**Reply by tube0013** [reply]
Can I send you a P7 based adaptor? I have some built that I'm testing now. BeagleBoard also has a P7 based board out <https://www.beagleboard.org/boards/beagleconnect-freedom>.

**Reply by Koenkk** [reply]
TI already send me this one: https://www.beagleboard.org/boards/beagleplay but AFAIK it works with the CC2652P7 firmware. In my prod setup I'm currently running the CC2652P7 launchpad with TX at 9.

**Reply by tube0013** [reply]
Okay - keep an open mind, the RF-Star module is used all over the place now, and is a drop in replacement for the P2 module - same footprint/pinout so simple switchover. I think 9 is fine for TX, but some people want the 20 ;)

Thanks!

**Reply by Koenkk** [reply]
The max that can be configured is 10:

<img width="770" alt="Screenshot 2023-08-09 at 19 54 43" src="https://github.com/Koenkk/Z-Stack-firmware/assets/2892853/4dc53437-87e6-43a5-a00f-da2fc7f864b4">


Hopefully TI will add 10-20 to their SDK
